core/src/main/java/org/infinispan/topology/CacheTopologyControlCommand.java

```diff
@@ -248,27 +253,127 @@ public byte getCommandId() {
    }
 
    @Override
-   public Object[] getParameters() {
-      return new Object[]{cacheName, (byte) type.ordinal(), sender, joinInfo, topologyId, rebalanceId, currentCH,
-            pendingCH, availabilityMode, actualMembers, throwable, viewId};
+   public void writeTo(ObjectOutput output) throws IOException {
+      MarshallUtil.marshallString(cacheName, output);
+      MarshallUtil.marshallEnum(type, output);
+      switch (type) {
```

Comment:
Selectively marshalling parameters probably makes sense more for commands that are used more often... for `CacheTopologyControlCommand`, I think serializing everything unconditionally would have been just fine.